clients/java/src/main/java/io/camunda/zeebe/client/api/command/PublishMessageCommandStep1.java

```diff
@@ -41,6 +41,18 @@ interface PublishMessageCommandStep2 {
      * @return the builder for this command
      */
     PublishMessageCommandStep3 correlationKey(String correlationKey);
+
+    /**
+     * Skip specifying a correlation key for the message.
+     *
+     * <p>This method allows the message to be published without a correlation key, making it
+     * suitable for scenarios where the correlation key is not necessary (e.g. for the message start
+     * event). When used, this will create a new process instance without checking for an active
+     * instance with the same correlation key.
+     *
+     * @return the builder for this command, continuing to the next step without a correlation key.
+     */
+    PublishMessageCommandStep3 skipCorrelationKey();
```

Comment:
For users, the `skip` wording might not be clear. It describes the act of skipping specifying the correlation key, but it needs to fit better from the perspective of building a publish message command. What does this mean for the message that is published? See example below:

```java
client
  .newPublishMessageCommand()
  .messageName("my_message")
  .skipCorrelationKey()
  .send()
  .join();
```

🔧 Instead, we could name this `noCorrelationKey()`. That would signal to users what the method is used for. They will publish a message with no correlation key. They'd choose between `correlationKey(String)` and `noCorrelationKey()`.

```java
client
  .newPublishMessageCommand()
  .messageName("my_message")
  .noCorrelationKey()
  .send()
  .join();
```

🔧 Another alternative is `withoutCorrelationKey()`. This is my favorite and was [suggested by someone else](https://github.com/camunda/zeebe/issues/13649#issuecomment-1826053751).

```java
client
  .newPublishMessageCommand()
  .messageName("my_message")
  .withoutCorrelationKey()
  .send()
  .join();
```

Reply:
`withoutCorrelationKey` +1

Reply:
Hi @korthout ! 
Thank you for your detailed and informative review! 🙇

I'm glad to hear that you liked the Javadoc on the interface. 😊

I've implemented your suggestion to improve the naming of the new method and have already rebased my changes into a single commit in my feature branch.
Thank you for guiding me in the right direction!

Could you please let me know if there are any other specific changes or steps left before merging this PR?
And how to proceed how to proceed with the merging process? (is it by adding `bors merge` comment to this PR?)

Reply:
Thanks for the changes @ce-dmelnych 💯 

👍 LGTM 

💭 We've been moving away from merging with bors to using GitHub Merge Queues (see https://github.com/camunda/zeebe/issues/12097). You can simply click the `Merge when ready` button below to queue your pull request for merging.

Reply:
Thank you, @korthout!
Probably I'm lacking some permissions.

This is what I see in the PR merge section:

<img width="923" alt="image" src="https://github.com/camunda/zeebe/assets/123628255/9b35fb9b-b552-49e6-8f72-7f8a3f08cc43">

Reply:
@abbasadel Can you give @ce-dmelnych access to the repo, for example by add @ce-dmelnych to `@camunda/zeebe-process-automation`?

Reply:
✅ Thanks @korthout.

Reply:
This issue was resolved!
Thank you,  @abbasadel for supporting me with it.

Reply:
@ce-dmelnych Feel free to merge 👍